setup.py

```diff
@@ -22,6 +21,24 @@
     'Topic :: Scientific/Engineering',
 ]
 
+class BuildExtWithNumpyCommand(build_ext):
```

Comment:
This stuff is much more complicated than what we had before, what is wrong with the simple two step install?

@ssanderson has expressed a desire to remove this stuff from zipline even.

Reply:
I had read your comment here https://github.com/quantopian/databazaar/pull/307#discussion_r144362567 as a request to move both numpy and Cython. I may have misread that, with you only meaning to move the `Cython` requirement.

Reply:
However, a point against the current version of `setup.py` which does the upfront import of `numpy` is that it prevents any extra tooling like `pip-compile`  from working in a fresh environment.

Reply:
you just need to install numpy, which you need to do anyways.